### ISSUE TITLE
 java.io.EOFException in SSEStream

### ISSUE THREAD

**Issue by dabitdev** [stellar/java-stellar-sdk#169]
Sometimes I get this issue when I am opening / closing the stream connection.

```
2019-01-25 14:28:47.765 23726-23817/com.blockeq.stellarwallet E/AndroidRuntime: FATAL EXCEPTION: OkHttp Dispatcher
    Process: com.blockeq.stellarwallet, PID: 23726
    java.lang.IllegalStateException: Failed 200
        at org.stellar.sdk.requests.SSEStream$StellarEventSourceListener.onFailure(SSEStream.java:176)
        at shadow.okhttp3.internal.sse.RealEventSource.processResponse(RealEventSource.java:83)
        at shadow.okhttp3.internal.sse.RealEventSource.onResponse(RealEventSource.java:54)
        at shadow.okhttp3.RealCall$AsyncCall.execute(RealCall.java:153)
        at shadow.okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: java.io.EOFException
        at shadow.okio.RealBufferedSource.require(RealBufferedSource.java:61)
        at shadow.okio.RealBufferedSource.readHexadecimalUnsignedLong(RealBufferedSource.java:304)
        at shadow.okhttp3.internal.http1.Http1Codec$ChunkedSource.readChunkSize(Http1Codec.java:469)
        at shadow.okhttp3.internal.http1.Http1Codec$ChunkedSource.read(Http1Codec.java:449)
        at shadow.okio.RealBufferedSource.indexOfElement(RealBufferedSource.java:394)
        at shadow.okio.RealBufferedSource.indexOfElement(RealBufferedSource.java:383)
        at shadow.okhttp3.internal.sse.ServerSentEventReader.processNextEvent(ServerSentEventReader.java:61)
        at shadow.okhttp3.internal.sse.RealEventSource.processResponse(RealEventSource.java:80)
        at shadow.okhttp3.internal.sse.RealEventSource.onResponse(RealEventSource.java:54) 
        at shadow.okhttp3.RealCall$AsyncCall.execute(RealCall.java:153) 
        at shadow.okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:764) 
```

**Comment by dabitdev** [comment]
closing and opening the stream 
I got this other one too
```
    Process: com.blockeq.stellarwallet, PID: 24221
    java.lang.IllegalStateException: Failed -1
        at org.stellar.sdk.requests.SSEStream$StellarEventSourceListener.onFailure(SSEStream.java:176)
        at shadow.okhttp3.internal.sse.RealEventSource.onFailure(RealEventSource.java:99)
        at shadow.okhttp3.RealCall$AsyncCall.execute(RealCall.java:161)
        at shadow.okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: java.io.IOException: Canceled
        at shadow.okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:120)
        at shadow.okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at shadow.okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at shadow.okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:200)
        at shadow.okhttp3.RealCall$AsyncCall.execute(RealCall.java:147)
        at shadow.okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:764) 
```

**Comment by bartekn** [comment]
This is fixed in [0.4.1](https://github.com/stellar/java-stellar-sdk/releases/tag/0.4.1).

**Comment by dabitdev** [comment]
I am already using 0.4.1
